go/service/main.go

```diff
@@ -619,30 +619,12 @@ func (d *Service) slowChecks() {
 }
 
 func (d *Service) tryGregordConnect() error {
-	// If we're logged out, LoggedInLoad() will return false with no error,
-	// even if the network is down. However, if we're logged in and the network
-	// is down, it will still return false, along with the network error. We
-	// need to handle that case specifically, so that we still start the gregor
-	// connect loop.
-	loggedIn, err := d.G().LoginState().LoggedInProvisioned(context.Background())
-	if err != nil {
-		// A network error means we *think* we're logged in, and we tried to
-		// confirm with the API server. In that case we'll swallow the error
-		// and allow control to proceed to the gregor loop. We'll still
-		// short-circuit for any unexpected errors though.
-		switch err.(type) {
-		case libkb.LoginStateTimeoutError, libkb.APINetError:
-			d.G().Log.Debug("Network/timeout error received from LoginState, continuing onward: %s", err)
-		default:
-			d.G().Log.Debug("Unexpected non-network error in tryGregordConnect: %s", err)
-			return err
-		}
-	} else if !loggedIn {
+	loggedIn := d.G().ActiveDevice.Valid()
+	if !loggedIn {
```

Comment:
I think this looks ok, but @mmaxim should look too.  This looks like biggest change in this PR so far...

Reply:
I haven't been following this too closely, so if this is basically equivalent as to before I guess it is ok.

Reply:
So I think before it could make a request to figure out if you are logged in, and now it only relies on ActiveDevice.

It's different, but I think it is ok.

That being said, it will be something to look out for once this is merged.

Reply:
doesn't this run after `tryLogin`?